### PR TITLE
Issue #2109: CLI should print a flie name where exception is happen

### PR DESCRIPTION
**Pull request by romani**

### CODE REVIEW

**Comment by romani** [comment]
build is failed, https://ci.appveyor.com/project/Checkstyle/checkstyle/build/2538/job/9t10fdi1rn7qrkbp#L627 but fix is obvious  - different file separators. I will fix it a bit later.

@mkordas , please confirm that you are ok with general approach for issue fix.

**Comment by romani** [comment]
additional fix is required to make
```
java -jar checkstyle-6.12-SNAPSHOT-all.jar -c 3.xml   3.java
Starting audit...
Exception happens during processing of 3.java
Cause: java.lang.NullPointerException
Checkstyle ends with 1 errors.
```
after additional commit

```
/var/tmp $ java -jar checkstyle-6.12-SNAPSHOT-all.jar -c 3.xml   3.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception happens during processing of 3.java
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:281)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:267)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:119)
Caused by: java.lang.NullPointerException
	at com.puppycrawl.tools.checkstyle.checks.coding.ModifiedControlVariableCheck.checkIdent(ModifiedControlVariableCheck.java:279)
	at com.puppycrawl.tools.checkstyle.checks.coding.ModifiedControlVariableCheck.visitToken(ModifiedControlVariableCheck.java:199)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:390)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:501)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:332)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:204)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:79)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:266)
	... 2 more
Checkstyle ends with 1 errors.
```

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 1]
```diff
@@ -577,4 +581,32 @@ public void testListFilesDirectoryWithNull() throws Exception {
         final List<File> result = (List<File>) method.invoke(null, fileMock);
         assertEquals(0, result.size());
     }
+
+    @Test
+    public void testFileReferenceDuringException() throws Exception {
+        exit.expectSystemExitWithStatus(-2);
+        final String cause = "Cause: com.puppycrawl.tools.checkstyle.api.CheckstyleException:"
```
Comment (mkordas) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Codacy Issue found: [Avoid unused local variables such as 'cause'.](https://www.codacy.com/app/checkstyle/checkstyle/file/1225112502/issues/source?bid=2193633&fileBranchId=2931618#l588)

**Comment by mkordas** [comment]
@romani 
I will look into it tomorrow, can you fix builds in the meantime?

**Comment by romani** [comment]
yes, I will do this in 4 hour.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 2]
```diff
@@ -577,4 +555,32 @@ public void testListFilesDirectoryWithNull() throws Exception {
...
```
Comment (mkordas) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Codacy Issue found: [Avoid unused local variables such as 'cause'.](https://www.codacy.com/app/checkstyle/checkstyle/file/1229260973/issues/source?bid=2193633&fileBranchId=2931618#l562)

**Comment by romani** [comment]
moved to #2487

**Pull request closed without merging**